### PR TITLE
added useApi as a custom hook

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

First task you told me to do (moving axios to UrlDataSection) has been made on branch fetching-Api.

I know... I know :) I should have first merged this branch after your approval. And then make next changes with axios. My mistake.

Anyway. I moved api to parent component URLDataSection as you told me. It was easier than I expected as I was pretty sure there going to be planty of mistakes after moving this whole logic to fetch data. But it went well.

So as it went so well and I still had some time in the morning I tried to write custom hook on my own since you said it will be our next thing to do.

And then again my mistake :( I created a custom hook and commited it stright to develop branch (oops..).

So it is already there on develop but I created also a new branch here so I can make commit again and so you can first check it. 

Please have a look.

Once again sorry for all this mess.

Regards,
Martin

### CODE REVIEW

**Review by AdamKniec — APPROVED** [review]

**Comment by marcinel123** [comment]
Hi Adam,

I changed useApi to useShorteningApi. It only makes api call and sets localStorage right now.

I moved useEffect with localStorage to UrlDataSection.

Also I changed some flexbox in FormElement so there are no longer two Error Paragraphs.

Regards,
Martin

File: app/src/api/useShorteningApi.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      ]);
+      localStorage.setItem(
+        "list",
+        JSON.stringify([
+          ...linksList,
+          { longLink: url, shortLink: response.data.result.short_link },
+        ])
+      );
+    } catch (err) {
+      if (err.response) {
```
Comment (AdamKniec) [inline app/src/api/useShorteningApi.js]: Is it possible for error not to have the response? Why is this check needed?

File: app/src/api/useShorteningApi.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import api from "./api";
+
+export const useShorteningApi = (url) => {
```
Comment (AdamKniec) [inline app/src/api/useShorteningApi.js:4]: 👏 👏 👏  Great job!

File: app/src/components/FormElement/FormElement.js [hunk 3]
```diff
@@ -32,20 +23,15 @@ export const FormElement = ({
           type="text"
           placeholder="Shorten a link here..."
           onChange={handleInput}
-          className={`${isVisible ? true : ""}`}
+          className={`${error ? true : ""}`}
         />
-        {isVisible ? (
+        {error ? (
```
Comment (AdamKniec) [inline app/src/components/FormElement/FormElement.js]: We Do not want to render empty string if there is no error. 

Did you try to {error && <ErrorParagraph>} ? You will avoid the ternary operator

**Review by AdamKniec — APPROVED** [review]
Good job

File: app/src/api/useShorteningApi.js (hunk 1 above)
Comment (marcinel123) [reply app/src/api/useShorteningApi.js]: fixed

File: app/src/api/useShorteningApi.js (hunk 2 above)
Comment (marcinel123) [reply app/src/api/useShorteningApi.js:4]: thx :)

File: app/src/components/FormElement/FormElement.js (hunk 3 above)
Comment (marcinel123) [reply app/src/components/FormElement/FormElement.js]: fixed

**Pull request merged**